HashRegistrarSimplified.sol

```diff
@@ -286,30 +318,23 @@ contract Registrar {
         if (address(bid) == 0 ) throw;
         sealedBids[seal] = Deed(0);
         bid.setOwner(_owner);
-        entry h = entries[_hash];
+        entry h = _entries[_hash];
         uint actualValue = min(_value, bid.balance);
-        bid.setBalance(actualValue);            
-
-        
-        if (h.status == Mode.Forbidden) {
-            /* 
-            If during the auction process someone reveals that the name being bidded
-            on is invalid (shorter than 7 chars) then all bids will be fully refunded
-            */
-            bid.closeDeed(1000);
-            BidRevealed(_hash, _owner, actualValue, 4);
-        } else if (bid.creationDate() > h.registrationDate - revealPeriod
-            || now > h.registrationDate 
-            || actualValue < minPrice) {
-            /* 
-            A penalty (99%) is applied for submitting unrevealed bids, which could otherwise
-            be used as a threat of revealing a bid higher than the second-highest 
-            bid, to extort the winner into paying them.
-            */
-            bid.closeDeed(10);
+        bid.setBalance(actualValue);
+
+        var auctionState = state(_hash);
+        if(auctionState == Mode.Owned) {
+            // Too late! Bidder loses their bid.
+            bid.closeDeed(0);
+            BidRevealed(_hash, _owner, actualValue, 1);
+        } else if(auctionState != Mode.Reveal) {
```

Comment:
This is another consequence of simplifying the code to be explicitly state-based; allowing reveal in multiple states would complicate the code. I don't think there's a compelling reason to complicate the code to allow early reveals.